sqlstring.js

```diff
@@ -0,0 +1,23 @@
+const sqlstring = require("sqlstring");
+const constants = require("./constants");
+
+function format(sql, values, dialect) {
+  if (dialect === constants.dialects.postgres) {
+    return sql.replace(/(\$\d+)/gi, binding => {
+      const index = parseInt(binding.slice(1));
+      const value = values[index - 1];
+
```

Comment:
Actually adding 

```
      if (value === null) {
        return null;
      }
```
Solves the nulling issue

Reply:
Good catch. Think the version of sqlstring might be a bit optimistic. I'll see if I can add a test for null columns.

Reply:
Added a fix very similar to what you suggested. Could you check if it works for you?

Reply:
I'm also looking into converting the prepared statement into something `knex.raw` can use. I'm a bit worried this approach might be vulnerable to SQL Injection

Reply:
I noticed that data-api-client supports named parameters. Would it be an option to rewrite the query to use named parameters instead of passing it as a raw sql string? Guess that would help with the sql injections? Guess we still would need separate postgres/mysql implementations for this though which is a pity.

Reply:
By importing knex

`const knexClient = require('knex')({ client: 'postgres' });`
and applying a few changes to the query
`const sql = knexClient.raw(obj.sql.replace(/`/g, '"').replace(/\$(\d+)/g, '?'), obj.bindings).toString();`

I was able to resolve the Date, null and issues with quotes within strings.

Reply:
This sounds promising.. I'll see if I can get this in today.

Reply:
This sounds promising.. I'll see if I can get this in today.

Reply:
Sorry for the delay.. I added this to the ma/sqlstring branch and will test it out before merging to the master of the PR-repo

Reply:
the replaces replace backticks ` with quotes " and then the other replace changes `$1, $2, $3` etc into `?`. I thought backticks were compatible with postgres but I was getting errors so maybe not. If there's a more elegant way to convert the query that would be nice. But after that it uses `knex.raw` so should avoid any possible SQL injections.